aviatrix/resource_aviatrix_firewall_instance.go

```diff
@@ -249,22 +249,21 @@ func resourceAviatrixFirewallInstanceCreate(d *schema.ResourceData, meta interfa
 	// gateway name. If there is an issue, we will just continue on without the additional
 	// validation.
 	var cloudType int
-	gw, err := client.GetGateway(&goaviatrix.Gateway{GwName: firewallInstance.GwName})
-	if err != nil {
-		log.Printf("[WARN] Could not get cloud_type from firenet_gw_name: %v", err)
+	if firewallInstance.GwName == "" {
```

Comment:
If firewallInstance.GwName == "", it could be user forgetting to set the value. So, if it is empty, need to confirm it is AWS GWLB Firenet; on the other hand, if it is not empty, need to confirm it is not AWS GWLB.

Reply:
I can try to get cloud_type from the vpc id

> on the other hand, if it is not empty, need to confirm it is not AWS GWLB.
I don't this is necessary. Backend will return an error if the VPC cannot support a firewall instance